pkg/mux/fragmentedConn.go

```diff
@@ -88,15 +88,23 @@ func (fc *fragmentedConnection) Write(b []byte) (n int, err error) {
 }
 
 func (fc *fragmentedConnection) Close() error {
+	// Stop the timer to prevent timer leak
+	if fc.isDead != nil {
+		fc.isDead.Stop()
```

Comment:
This is not a leak. 

Before go1.23 use of the `timer.After(...)` function would result in a GC memory leak until the timer was fired. 
Here we're using `AfterFunc`, which actually uses a different internal structure, and lacks the warning that it leaks memory. 

Additionally we use go1.23.1 which no longer has this issue for its `time.After(...)` func.